### PR TITLE
Enable more linters and update config

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Update the linter config and fix a bunch of lints.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes-sigs/cri-tools-maintainers PTAL

File: cmd/crictl/config.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ CRICTL OPTIONS:
 			}
 			return common.WriteConfig(config, configFile)
 		} else if c.Bool("list") {
-			display := newTableDisplay(20, 1, 3, ' ', 0)
```
Comment (SergeyKanzhelev) [inline cmd/crictl/config.go:109]: I wonder what linter rule did it break?

File: .golangci.yml [hunk 2]
```diff
@@ -38,81 +55,69 @@ linters:
...
+    - noctx
+    - nolintlint
     - nosprintfhostport
     - perfsprint
     - prealloc
     - predeclared
     - promlinter
     - protogetter
     - reassign
+    - rowserrcheck
```
Comment (kwilczynski) [inline .golangci.yml:67]: Not sure if we need to enable this one?

File: .golangci.yml [hunk 3]
```diff
@@ -38,81 +55,69 @@ linters:
...
     - sloglint
     - spancheck
+    - sqlclosecheck
```
Comment (kwilczynski) [inline .golangci.yml:70]: Ditto as https://github.com/kubernetes-sigs/cri-tools/pull/1496/files#r1680376638.

Comment (saschagrunert) [reply .golangci.yml:70]: I'd say leave it enabled if we either add code which the linter may catch in the future.

File: .golangci.yml (hunk 2 above)
Comment (saschagrunert) [reply .golangci.yml:67]: I'd say leave it enabled if we either add code which the linter may catch in the future.

File: cmd/crictl/config.go (hunk 1 above)
Comment (saschagrunert) [reply cmd/crictl/config.go:109]: It's the [gosimple](https://golangci-lint.run/usage/linters/#gosimple) linter, but I could work around by introducing a `newDefaultTableDisplay` method and reusing it everywhere where it's needed.

**Pull request merged**